In [1]:
import numpy as np
import pandas as pd
import sklearn
import seaborn as sns
import matplotlib.pyplot as plt
sns.set()


In [2]:
df=pd.read_csv('customer_retention.csv')
df=pd.DataFrame(df) 

In [3]:
df.head()

1Gender of respondent  2 How old are you?   \
0                      0                    3   
1                      1                    2   
2                      1                    2   
3                      0                    2   
4                      1                    2   

  3 Which city do you shop online from?  \
0                                 Delhi   
1                                 Delhi   
2                         Greater Noida   
3                               Karnal    
4                            Bangalore    

   4 What is the Pin Code of where you shop online from?  \
0                                             110009       
1                                             110030       
2                                             201308       
3                                             132001       
4                                             530068       

   5 Since How Long You are Shopping Online ?  \
0                                           5   
1                                           5   
2                                           4   
3                                           4   
4                                           3   

   6 How many times you have made an online purchase in the past 1 year?  \
0                                                  4                       
1                                                  5                       
2                                                  5                       
3                                                  1                       
4                                                  2                       

   7 How do you access the internet while shopping on-line?  \
0                                                  4          
1                                                  2          
2                                                  3          
3                                                  3          
4                                                  2          

   8 Which device do you use to access the online shopping?  \
0                                                  3          
1                                                  1          
2                                                  1          
3                                                  1          
4                                                  1          

   9 What is the screen size of your mobile device?\t\t\t\t\t\t                                          \
0                                                  5                                                      
1                                                  2                                                      
2                                                  4                                                      
3                                                  4                                                      
4                                                  2                                                      

   10 What is the operating system (OS) of your device?\t\t\t\t                                            \
0                                                  1                                                        
1                                                  3                                                        
2                                                  2                                                        
3                                                  3                                                        
4                                                  3                                                        

   ...  Longer time to get logged in (promotion, sales period)  \
0  ...                                          Amazon.in        
1  ...                            Amazon.in, Flipkart.com        
2  ...                                         Myntra.com        
3  ...          

In [6]:
df.shape            # dimension of data set is 269 rows and 71 columns.

(269, 71)

In [9]:
df.keys()           # names of columns

Index(['1Gender of respondent', '2 How old are you? ',
       '3 Which city do you shop online from?',
       '4 What is the Pin Code of where you shop online from?',
       '5 Since How Long You are Shopping Online ?',
       '6 How many times you have made an online purchase in the past 1 year?',
       '7 How do you access the internet while shopping on-line?',
       '8 Which device do you use to access the online shopping?',
       '9 What is the screen size of your mobile device?\t\t\t\t\t\t                                        ',
       '10 What is the operating system (OS) of your device?\t\t\t\t                                          ',
       '11 What browser do you run on your device to access the website?\t\t\t                                                          ',
       '12 Which channel did you follow to arrive at your favorite online store for the first time?                                                                   ',
       '13 After first visit, how 

In [10]:
df.info()      # here we have integer and objects present in data

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 269 entries, 0 to 268
Data columns (total 71 columns):
 #   Column                                                                                                                                                                                                          Non-Null Count  Dtype 
---  ------                                                                                                                                                                                                          --------------  ----- 
 0   1Gender of respondent                                                                                                                                                                                           269 non-null    int64 
 1   2 How old are you?                                                                                                                                                                    

In [12]:
df.isnull().sum()     # we do not have any null values present in data set.

1Gender of respondent                                                   0
2 How old are you?                                                      0
3 Which city do you shop online from?                                   0
4 What is the Pin Code of where you shop online from?                   0
5 Since How Long You are Shopping Online ?                              0
                                                                       ..
Longer delivery period                                                  0
Change in website/Application design                                    0
Frequent disruption when moving from one page to another                0
Website is as efficient as before                                       0
Which of the Indian online retailer would you recommend to a friend?    0
Length: 71, dtype: int64

In [13]:
df.describe()

1Gender of respondent  2 How old are you?   \
count             269.000000           269.000000   
mean                0.669145             2.959108   
std                 0.471398             1.066012   
min                 0.000000             1.000000   
25%                 0.000000             2.000000   
50%                 1.000000             3.000000   
75%                 1.000000             4.000000   
max                 1.000000             5.000000   

       4 What is the Pin Code of where you shop online from?  \
count                                         269.000000       
mean                                       220465.747212       
std                                        140524.341051       
min                                        110008.000000       
25%                                        122018.000000       
50%                                        201303.000000       
75%                                        201310.000000       
max                                        560037.000000       

       5 Since How Long You are Shopping Online ?  \
count                                  269.000000   
mean                                     3.524164   
std                                      1.436586   
min                                      1.000000   
25%                                      3.000000   
50%                                      4.000000   
75%                                      5.000000   
max                                      5.000000   

       6 How many times you have made an online purchase in the past 1 year?  \
count                                         269.000000                       
mean                                            2.672862                       
std                                             1.651788                       
min                                             1.000000                       
25%                                             1.000000                       
50%                                             2.000000                       
75%                                             4.000000                       
max                                             5.000000                       

       7 How do you access the internet while shopping on-line?  \
count                                         269.000000          
mean                                            3.260223          
std                                             1.135887          
min                                             2.000000          
25%                                             2.000000          
50%                                             3.000000          
75%                                             5.000000          
max                                             5.000000          

       8 Which device do you use to access the online shopping?  \
count                                         269.000000          
mean                                            1.676580          
std                                             0.843904          
min                                             1.000000          
25%                                             1.000000          
50%                                             1.000000          
75%                                             2.000000          
max                                             4.000000          

       9 What is the screen size of your mobile device?\t\t\t\t\t\t                                          \
count                                         269.000000                                                      
mean                                            4.282528                                                      
std                                             0.923426                                                      
min                                             2.000000                                             

Above we have details of data set like mean, std, max and minumum values of each columns present in data set.

In [22]:
df['1Gender of respondent'].value_counts()      sns.countplot(x = "1Gender of responden", data = df)

ValueError: Could not interpret input '1Gender of responden'

In [18]:
df['3 Which city do you shop online from?'].value_counts()  #


Delhi            58
Greater Noida    43
Noida            40
Bangalore        37
Karnal           27
Ghaziabad        18
Solan            18
Gurgaon          12
Merrut            9
Moradabad         5
Bulandshahr       2
Name: 3 Which city do you shop online from?, dtype: int64

We understand the online customer shopping based according to city from above data.

In [19]:
df['4 What is the Pin Code of where you shop online from?'].value_counts() 

201308    38
132001    19
201310    18
110044    16
173212     9
250001     9
173229     9
560037     8
132036     8
560010     8
122018     8
110011     7
201306     7
110008     7
110014     6
110018     6
244001     5
201312     5
530068     5
201305     5
201009     5
201008     5
201005     4
201001     4
201304     4
110039     4
110030     4
110009     4
560018     4
560003     4
122009     4
560002     4
110042     4
201303     4
560013     3
203001     2
560001     1
203207     1
203202     1
Name: 4 What is the Pin Code of where you shop online from?, dtype: int64

Maximum shopping is done from pin code 201308    

In [20]:
df['6 How many times you have made an online purchase in the past 1 year?'].value_counts() 

1    114
4     63
5     53
2     29
3     10
Name: 6 How many times you have made an online purchase in the past 1 year?, dtype: int64

Above data suggest that the users are using the website for once for online shopping the count is 114, there is significant number of customers that are loyal to wesites.
Websites should try to focus on user which are shopping for first time from there website. Companies or website should plan and roll out attarctive schemes for first time shoppers.


In [21]:
df['7 How do you access the internet while shopping on-line?'].value_counts() 

3    118
2     76
5     71
4      4
Name: 7 How do you access the internet while shopping on-line?, dtype: int64

In [ ]:
df['2 How old are you?'].value_counts() 

In [ ]:
df['2 How old are you?'].value_counts() 

In [ ]:
df['2 How old are you?'].value_counts() 

In [ ]:
df['2 How old are you?'].value_counts() 

In [ ]:
df['2 How old are you?'].value_counts() 

In [ ]:
df['2 How old are you?'].value_counts() 

In [ ]:
df['2 How old are you?'].value_counts() 

In [ ]:
df['2 How old are you?'].value_counts() 

In [ ]:
df['2 How old are you?'].value_counts() 

In [ ]:
df['2 How old are you?'].value_counts() 

In [ ]:
df['2 How old are you?'].value_counts() 

In [ ]:
df['2 How old are you?'].value_counts() 

In [ ]:
df['2 How old are you?'].value_counts() 

In [ ]:
df['2 How old are you?'].value_counts() 

In [ ]:
df['2 How old are you?'].value_counts() 

In [ ]:
df['2 How old are you?'].value_counts() 

In [ ]:
df['2 How old are you?'].value_counts() 

In [ ]:
df['2 How old are you?'].value_counts() 

In [ ]:
df['2 How old are you?'].value_counts() 

In [ ]:
df['2 How old are you?'].value_counts() 

In [ ]:
df['2 How old are you?'].value_counts() 

In [ ]:
df['2 How old are you?'].value_counts() 

In [ ]:
df['2 How old are you?'].value_counts() 

In [ ]:
df['2 How old are you?'].value_counts() 

In [ ]:
df['2 How old are you?'].value_counts() 

In [ ]:
df['2 How old are you?'].value_counts() 

In [ ]:
df['2 How old are you?'].value_counts() 

In [ ]:
df['2 How old are you?'].value_counts() 

In [ ]:
df['2 How old are you?'].value_counts() 

In [ ]:
df['2 How old are you?'].value_counts() 

In [ ]:
df['2 How old are you?'].value_counts() 

In [ ]:
df['2 How old are you?'].value_counts() 

In [ ]:
df['2 How old are you?'].value_counts() 

In [ ]:
df['2 How old are you?'].value_counts() 

In [ ]:
df['2 How old are you?'].value_counts() 

In [ ]:
df['2 How old are you?'].value_counts() 

In [ ]:
df['2 How old are you?'].value_counts() 

In [ ]:
df['2 How old are you?'].value_counts() 

In [ ]:
df['2 How old are you?'].value_counts() 

In [ ]:
df['2 How old are you?'].value_counts() 